pkg/kubelet/kuberuntime/podsandbox.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/pkg/kubelet/util/format"
+)
+
+// getKubeletPodSandboxes lists all (or just the running) pod sandboxes managed by kubelet.
+func (m *kubeGenericRuntimeManager) getKubeletPodSandboxes(all bool) ([]*runtimeApi.PodSandbox, error) {
+	var filter *runtimeApi.PodSandboxFilter
+	if !all {
+		readyState := runtimeApi.PodSandBoxState_READY
+		filter = &runtimeApi.PodSandboxFilter{
+			State: &readyState,
+		}
+	}
+
+	resp, err := m.runtimeService.ListPodSandbox(filter)
+	if err != nil {
+		glog.Errorf("ListPodSandbox failed: %v", err)
+		return nil, err
+	}
+
+	result := []*runtimeApi.PodSandbox{}
+	for _, s := range resp {
+		if len(s.GetName()) == 0 {
+			continue
+		}
+
+		_, _, _, _, _, err := parseContainerName(s.GetName())
```

Comment:
nit: Put the logic into a function like `IsPodSandboxManagedByKubelet`, or else the `_, _, _, _, _, err` logic seems a little unclear here. :)